aioesphomeapi/connection.py

```diff
@@ -725,86 +729,72 @@ def _report_fatal_error(self, err: Exception) -> None:
         self._set_connection_state(ConnectionState.CLOSED)
         self._cleanup()
 
-    def _process_packet_factory(self) -> Callable[[int, bytes], None]:
+    def _process_packet(self, msg_type_proto: _int, data: _bytes) -> None:
```

Comment:
Since cython can nearly avoid the cost of `self` it didn't make sense to make it a factory anymore